IO/NativeSignalHandlers/UnixHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Mono.Unix;
+using Mono.Unix.Native;
+using System.Threading.Tasks;
+
+namespace LocalAdmin.V2.IO.NativeSignalHandlers
+{
+    /// <summary>
+    ///     Native signal processing on Unix systems.
+    /// </summary>
+    internal sealed class UnixHandler : INativeSignalHandler
+    {
+        public static readonly UnixHandler Handler;
+        private static readonly UnixSignal[] signals;
+
+        static UnixHandler()
+        {
+            Handler = new UnixHandler();
+            signals = new UnixSignal[]
+            {
+                new UnixSignal(Signum.SIGINT),  // CTRL + C pressed
+                new UnixSignal(Signum.SIGTERM), // Sending KILL
+                new UnixSignal(Signum.SIGUSR1),
+                new UnixSignal(Signum.SIGUSR2),
+                new UnixSignal(Signum.SIGHUP)   // Terminal is closed
+            };
+        }
+
+        private UnixHandler() { }
+
+        public void Setup()
+        {
+            Task.Run(() =>
```

Comment:
Wouldn't a thread be better here?

Reply:
What for? We don't have real work, it's a task that awaits its awakening which is equivalent `await Task.Delay(10);` and is performed in the thread pool.

Reply:
Tasks are intended to be used for short running code, this waits for the whole lifetime of the application, it's just a bad practice to use a task for here, especially since it doesn't do anything async

Reply:
Ok I'll fix it

Reply:
Resolved in https://github.com/northwood-studios/LocalAdmin-V2/pull/4/commits/d5ef912f0389b423de4daec25e15bbb4fcf6543a
And I also made the thread safe `LocalAdmin::Exit` method, but I’m not in a hurry to resolve this conversation, maybe I did something not quite right and you still have questions.